### ISSUE TITLE
Using with a SurfaceView?

### ISSUE THREAD

**Issue by tmm1** [Dimezis/BlurView#60]
I'd like to use the BlurView on top of a SurfaceView (which is playing video), and show a dynamic blur effect based on the underlying video. Can this work with BlurView, or is the blur generated only once?

I have a simple FrameLayout which contains my SurfaceView and my BlurView. However I must not be configuring it correctly because I don't see any blur effects, just a simple background with my configured blurOverlayColor.

**Comment by Dimezis** [comment]
Hi, it's not possible right now, check out https://github.com/Dimezis/BlurView/issues/56 for more info

**Comment by tmm1** [comment]
Thanks for the quick response and link.

Can you tell me which API in 23+ you think can be used for this purpose?

**Comment by Dimezis** [comment]
https://developer.android.com/reference/android/view/Surface.html#lockHardwareCanvas() This one, but it requires  a completely different approach for BlurView implementation. Check out texture view branch if interested

**Comment by Dimezis** [comment]
I just sketched a POC locally, trying to blur the camera preview. Small clarification, it's possible to blur a TextureView using `lockHardwareCanvas()`, but not the SurfaceView, because of the way SurfaceView performs drawing. If you are fine with switching to TextureView to play video, and with API 23+ I can push the sketch I made and expose some kind of API to switch to hardware canvas.

There's one issue though, the hardware canvas' performance is sometimes worse than the regular canvas on some of devices I tested it. 
So I'm not sure if this feature is usable with the current state of things on Android =\
